docs/concepts/validators.md

```diff
@@ -1,635 +1,575 @@
+[annotated pattern](./fields.md#the-annotated-pattern) or using the
+[`field_validator()`][pydantic.field_validator] decorator, applied on a [class method][classmethod]:
 
-from typing_extensions import Annotated
+[](){#field-after-validator}
 
-from pydantic import BaseModel, ValidationError
-from pydantic.functional_validators import AfterValidator
+- __*After* validators__: they run after Pydantic's internal validation. They are generally more type safe and thus easier to implement.
 
+    === "Annotated pattern"
 
-def check_squares(v: int) -> int:
-    assert v**0.5 % 1 == 0, f'{v} is not a square number'
-    return v
+        Here is an example of a validator performing a validation check, and returning the value unchanged.
 
+        ```python
+        from typing_extensions import Annotated
 
-def double(v: Any) -> Any:
-    return v * 2
+        from pydantic import AfterValidator, BaseModel, ValidationError
 
 
-MyNumber = Annotated[int, AfterValidator(double), AfterValidator(check_squares)]
+        def is_even(value: int) -> int:
+            if value % 2 == 1:
+                raise ValueError(f'{value} is not an even number')
+            return value  # (1)!
 
 
-class DemoModel(BaseModel):
-    number: List[MyNumber]
+        class Model(BaseModel):
+            number: Annotated[int, AfterValidator(is_even)]
 
 
-print(DemoModel(number=[2, 8]))
-#> number=[4, 16]
-try:
-    DemoModel(number=[2, 4])
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for DemoModel
-    number.1
-      Assertion failed, 8 is not a square number
-    assert ((8 ** 0.5) % 1) == 0 [type=assertion_error, input_value=4, input_type=int]
-    """
-```
+        try:
+            Model(number=1)
+        except ValidationError as err:
+            print(err)
+            """
+            1 validation error for Model
+            number
+              Value error, 1 is not an even number [type=value_error, input_value=1, input_type=int]
+            """
+        ```
 
-In this example we used some type aliases (`MyNumber = Annotated[...]`).
-While this can help with legibility of the code, it is not required, you can use `Annotated` directly in a model field type hint.
-These type aliases are also not actual types but you can use a similar approach with `TypeAliasType` to create actual types.
-See [Custom Types](../concepts/types.md#custom-types) for a more detailed explanation of custom types.
+        1. Note that it is important to return the validated value.
 
-It is also worth noting that you can nest `Annotated` inside other types.
-In this example we used that to apply validation to the inner items of a list.
-The same approach can be used for dict keys, etc.
+    === "Decorator"
 
-### Before, After, Wrap and Plain validators
+        Here is an example of a validator making changes to the validated value (no exception is raised),
+        this time using the [`field_validator()`][pydantic.field_validator] decorator.
```

Comment:
As mentioned on our call, I think the examples provided in the tabs should be analogous so that it's easy to understand the parallels between annotated validators and validators via decoration.

That being said, I think both concepts here are important, so I don't think we should delete this example. Happy to chat more about where it might go.

Reply:
What if we added a collapsable section with another example so that it stays localized on this page but doesn't consume too much space?